This is great, thanks for doing this. There seems to be an issue though. I tried running it on this fib program:

```
def fib(n) {
  if (n < 2) {
    return 1;
  }
  return fib(n-1) + fib(n-2);
}
fib(15);
```

By saying `./bish -r fib.bish`. And I get the following bash error: `bash: line 7: syntax error: unexpected end of file`. When I run `./bish fib.bish | bash` I don't see that error. Could you look into that before I merge this?

Additionally, I get a compiler warning (clang) with your changes:

```
src/bish.cpp:83:14: warning: using the result of an assignment as a condition
      without parentheses [-Wparentheses]
    while (n = is.readsome(buf, sizeof(buf))) {
```

What you meant is perfectly clear, but to silence the warning could you just add parens around that assignment?

--- a/src/bish.cpp
+++ b/src/bish.cpp
@@ -80,7 +80,7 @@ void run_on_bash(std::istream &is) {
     FILE *bash = popen("bash", "w");
     char buf[4096];
     int n;
-    while (n = is.readsome(buf, sizeof(buf))) {
+    while ((n = is.readsome(buf, sizeof(buf)))) {
         fwrite(buf, 1, n, bash);
     }
 

--- a/src/bish.cpp
+++ b/src/bish.cpp
@@ -79,10 +79,11 @@ void compile_to_bash(std::ostream &os, Bish::Module *m) {
 void run_on_bash(std::istream &is) {
     FILE *bash = popen("bash", "w");
     char buf[4096];
-    while (is.read(buf, sizeof(buf))) {
+
+    do {
+        is.read(buf, sizeof(buf));
         fwrite(buf, 1, is.gcount(), bash);
-    }
-    fwrite(buf, 1, is.gcount(), bash);
+    } while (is.gcount() > 0);
 
     fflush(bash);
     pclose(bash);

--- a/src/bish.cpp
+++ b/src/bish.cpp
@@ -79,10 +79,10 @@ void compile_to_bash(std::ostream &os, Bish::Module *m) {
 void run_on_bash(std::istream &is) {
     FILE *bash = popen("bash", "w");
     char buf[4096];
-    int n;
-    while ((n = is.readsome(buf, sizeof(buf)))) {
-        fwrite(buf, 1, n, bash);
+    while (is.read(buf, sizeof(buf))) {
+        fwrite(buf, 1, is.gcount(), bash);
     }
+    fwrite(buf, 1, is.gcount(), bash);
 
     fflush(bash);
     pclose(bash);

